python/lsst/pipe/tasks/calibrate.py

```diff
@@ -319,14 +329,18 @@ def __init__(self, butler=None, refObjLoader=None, icSourceSchema=None, **kwargs
             self.makeSubtask('applyApCorr', schema=self.schema)
         self.makeSubtask('catalogCalculation', schema=self.schema)
 
-        if self.config.doAstrometry or self.config.doPhotoCal:
-            if refObjLoader is None:
-                self.makeSubtask('refObjLoader', butler=butler)
-                refObjLoader = self.refObjLoader
+        if self.config.doAstrometry:
+            if astromRefObjLoader is None:
+                self.makeSubtask('astromRefObjLoader', butler=butler)
+                refObjLoader = self.astromRefObjLoader
             self.pixelMargin = refObjLoader.config.pixelMargin
             self.makeSubtask("astrometry", refObjLoader=refObjLoader, schema=self.schema)
```

Comment:
I couldn't find an immediate way around the intermediate variable.  I just changed `refObjLoader` -> `astromRefObjLoader`.